etcdserver/server.go

```diff
@@ -247,7 +247,7 @@ func NewServer(cfg *ServerConfig) (*EtcdServer, error) {
 			var err error
 			str, err = discovery.JoinCluster(cfg.DiscoveryURL, cfg.DiscoveryProxy, m.ID, cfg.InitialPeerURLsMap.String())
 			if err != nil {
-				return nil, err
+				return nil, fmt.Errorf("error joining discovery cluster (%v)", err)
```

Comment:
yes. then the String.contains part will be more clear. Or we can actually have a function called IsDiscoveryError in discovery pkg.